thriftcli/thrift_cli.py

```diff
@@ -89,16 +84,17 @@ def cleanup(self, remove_generated_src=False):
     @classmethod
     def transform_output(cls, result, return_json=False):
         if return_json:
-            result = json.dumps(result, default=cls._default_json_handler, sort_keys=True, indent=4, separators=(',', ': '))
+            result = _dump_json(result)
         return result
 
-    @classmethod
-    def _default_json_handler(cls, obj):
-        if isinstance(obj, set):
-            return list(obj)
-        else:
-            return obj.__dict__
+def _dump_json(obj):
+    return json.dumps(obj, default=_default_json_handler, sort_keys=True, indent=4, separators=(',', ': '))
 
+def _default_json_handler(obj):
```

Comment:
Overall this PR looks right to me.

I think down the line this method may need to become recursive. (nested thrift objects). But that problem already existed It think.

Reply:
I believe that since this just returns a "mappable" object, the same default handler will be used then down the line. I've used it to dump nested sets and seems to be working fine.